nailgun/entities.py

```diff
@@ -8132,3 +8132,149 @@ def __init__(self, server_config=None, **kwargs):
+                    "build_exited",
+                    "content_view_created",
+                    "content_view_destroyed",
+                    "content_view_updated",
+                    "domain_created",
+                    "domain_destroyed",
+                    "domain_updated",
+                    "host_created",
+                    "host_destroyed",
+                    "host_updated",
+                    "hostgroup_created",
+                    "hostgroup_destroyed",
+                    "hostgroup_updated",
+                    "model_created",
+                    "model_destroyed",
+                    "model_updated",
+                    "status_changed",
+                    "subnet_created",
+                    "subnet_destroyed",
+                    "subnet_updated",
+                    "user_created",
+                    "user_destroyed",
+                    "user_updated",
+                ],
+            ),
+            'http_content_type': entity_fields.StringField(),
+            'http_method': entity_fields.StringField(
+                choices=[
+                    "POST",
+                    "GET",
+                    "PUT",
+                    "DELETE",
+                    "PATCH",
+                ]
+            ),
+            'password': entity_fields.StringField(),
+            'proxy_authorization': entity_fields.BooleanField(),
+            'target_url': entity_fields.URLField(required=True),
+            'user': entity_fields.StringField(),
+            'verify_ssl': entity_fields.BooleanField(),
+            'webhook_template': entity_fields.OneToOneField(WebhookTemplate),
+        }
+        self._meta = {
+            'api_path': 'api/webhooks',
+        }
+        super().__init__(server_config, **kwargs)
+
+    def create(self, create_missing=None):
+        """
+        Create WebhookTemplate properly when creating a parent Webhook instance
+        """
+        original_return = super().create(create_missing)
+        webhook_template_id = original_return.webhook_template.id
+        original_return.webhook_template = WebhookTemplate(
+            self._server_config, id=webhook_template_id
+        ).read()
+        return original_return
+
+    def read(self, entity=None, attrs=None, ignore=None, params=None):
+        """Ignore auth properties on read"""
+        if ignore is None:
+            ignore = set()
+        ignore.add('password')
+        ignore.add('proxy_authorization')
+        return super().read(entity, attrs, ignore, params)
+
+
+class WebhookTemplate(
+    Entity,
+    EntityCreateMixin,
+    EntityDeleteMixin,
+    EntityReadMixin,
+    EntitySearchMixin,
+    EntityUpdateMixin,
+):
+    """A representation of a WebhookTemplate entity."""
+
+    def __init__(self, server_config=None, **kwargs):
+        self._fields = {
+            'name': entity_fields.StringField(
+                required=True, str_type='alpha', length=(6, 12), unique=True
```

Comment:
Why would you want to limit this ? Is this asked by that entity ?

If you create this entity with `valid_data_list` parameters in robottelo tests it will fail ! I will recommand you to not to limit !

Reply:
Same for Webhook entity above !